include/deal.II/lac/arpack_solver.h

```diff
@@ -371,6 +431,13 @@ class ArpackSolver : public Subscriptor
    */
   double sigmai;
 
+  /**
+   * Value of the BMAT argument in ARPACK which is used to specify the type of
+   * the matrix B that defines the semi-inner product for the operator OP.
+   *   - BMAT = 'I' -> standard eigenvalue problem $A*x = \lambda *x$.
+   *   - BMAT = 'G' -> generalized eigenvalue problem $A*x = \lambda *B*x$.
+   */
+  char eigenvalue_problem_type = 'G';
```

Comment:
i have nit-picking comment: this is really just a switch so that we can run through the same code for both standard and generalized one. I don't think it belongs to the class, how about adding a `private:` solve (current one, most general one) and call it from two public solves for standard and generalized eigenproblems?

Reply:
I don't see how it gains us anything. We are going from two functions to three and I will need to pass the parameter around.

Reply:
> I don't see how it gains us anything. We are going from two functions to three and I will need to pass the parameter around.

It does not gain us anything, I personally find this more clear. Let's see what others have to say @dealii/dealii ping.

Reply:
See my comments below.